tasties_app/models.py

```diff
@@ -1,58 +1,57 @@
-from enum import Enum
-
-from django.contrib.auth.models import User
-from django.core.validators import (
-    MaxValueValidator,
-    MinLengthValidator,
-    MinValueValidator,
-)
 from django.db import models
 from django.utils import timezone
+from django.contrib.auth.models import User
+from django.core.validators import MinValueValidator, MinLengthValidator, MaxValueValidator
+from datetime import date
+from django.core.exceptions import ValidationError
 
 
 class Category(models.Model):
     category_name = models.CharField(max_length=16, unique=True)
 
 
 class Recipe(models.Model):
-    title = models.CharField(
-        max_length=64, validators=[MinLengthValidator(1)], unique=True
-    )
+    title = models.CharField(max_length=64, validators=[MinLengthValidator(1)], unique=True)
     author_id = models.ForeignKey(User, on_delete=models.CASCADE)
     categories = models.ManyToManyField(Category)
     description = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
     directions = models.CharField(max_length=65536, validators=[MinLengthValidator(1)])
     publication_date = models.DateField(default=timezone.now)
     minutes_to_make = models.IntegerField(validators=[MinValueValidator(1)])
-    recipe_picture = models.ImageField(upload_to="recipe_pictures")
+    recipe_picture = models.ImageField(upload_to='recipe_pictures')
 
+    def __str__(self):
+        return self.title
 
-class Ingredient(models.Model):
-    class UnitChoices(models.TextChoices, Enum):
-        WHOLE = "Whole"
-        FLOZ = "Fluid Ounce"
-        TSP = "Tea Spoon"
-        OZ = "Ounce"
-        CUP = "Cup"
-        GRAM = "Gram"
-        ML = "Milliliter"
 
+class Ingredient(models.Model):
     recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
-    amount = models.DecimalField(
-        max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)]
-    )
-    measurement_unit = models.CharField(
-        max_length=11, choices=UnitChoices.choices, default=UnitChoices.WHOLE
-    )
+    amount = models.DecimalField(max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)])
+    measurement_unit = models.TextChoices('measurement_unit', 'ML FLOZ TSP OZ CUP GRAM WHOLE')
```

Comment:
Why did you remove the class UnitChoices and switched to this?

Reply:
@WaseemAbuSobe did you mark this as resolved by mistake?
I still don't understand.

Reply:
@liornoy 
sorry. is was mistake now I fix.